e2e/vmss.go

```diff
@@ -0,0 +1,165 @@
+package e2e_test
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	mrand "math/rand"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute"
+	"golang.org/x/crypto/ssh"
+)
+
+func createVMSSWithPayload(ctx context.Context, r *mrand.Rand, cloud *azureClient, location, resourceGroupName, name, subnetID string, customData, cseCmd string, mutator func(*armcompute.VirtualMachineScaleSet)) (sshPrivateKey []byte, e error) {
```

Comment:
I might be blind, but is 'r' ever used in here?

Reply:
fixed -- using the global func without a seed is deprecated.